clueless/state.py

```diff
@@ -41,33 +51,63 @@ def sock_addr(self) -> str:
 
     @property
     def is_registered(self) -> bool:
-        return self.character is not None
+        return self.state is UserState.REGISTERED
 
     async def dispatch_message(self, message):
+        print("Dispatch_message function:")
         if isinstance(message, CreateGame):
-            self.logger.debug(f"Received crate game")
-            game = GameState()
-            id = game.id
-            self.logger.debug(f'Created game {id}')
-            GAMES[id] = game
-            await GAMES[id].add_user(self)
+            if self.game is None:                       # If this player instance is not in a game
+                self.game = GameState()                 # Create an instance of GameState class, associate it with Player instance
+                id = self.game.id                            
+                self.logger.debug(f'Created game {id}')
+                GAMES[id] = self.game                   # Place in a dictionary of games
+                await GAMES[id].add_user(self)          # Add user into recently created class
+            else:
+                self.send_message(Status("Player is already in a game")) # RuntimeWarning: coroutine 'Player.send_message' was never awaited
+        
         elif isinstance(message, JoinGame):
-            self.logger.debug(f'Joining game {message.id}')
-            await GAMES[message.id].add_user(self)
+            print("JoinGame instance")
+            if self.game is None:                           # Check whether this player instance is assocaited with a game
+
+                self.logger.debug(f'Player Validated')      
+                self.game = GAMES[message.id]               # Instance of player gets its own game association
+
+                if self.game.id in GAMES:                   # Check whether the game.id exists in dictionary
+                    self.logger.debug(f'Game {message.id} Validated')
+                    await GAMES[message.id].add_user(self)
+                else:
+                    self.logger.debug(f'Game ID {message.id} DOES NOT exist') # RuntimeWarning: coroutine 'Player.send_message' was never awaited
+                    self.send_message(Status("Game ID does NOT exist"))
+            else:
+                self.logger.debug(f'Player is in a game')
+                self.send_message(Status("Player is already in a game"))
+                
         elif isinstance(message, Register):
-            self.logger.debug(f'Registering as {message.display_name} with {message.character}')
-            self.character = message.character
-            self.display_name = message.display_name
-            await self.game.register_user(self, message)
+            print("Register instance")
+            if self.game.id in GAMES and not self.is_registered:    # Not quite sure if this is enough validation to say a player is in a certain game
```

Comment:
I think `if not self.is_registered` would be sufficient here. If the player has a value for `self.game` then `self.game.id` will always be in `GAMES`. If they aren't in a game, `self.game` will be `None` and this line would throw an exception.